firmware/library/L3_HAL/apds_9960.hpp

```diff
@@ -1,20 +1,444 @@
+// Low level Gesture Driver for setting up APDS-9960 device
+// Usage:
+//  Apds9960 sensor;
+//  sensor.Initialize();
+//  sensor.EnableGesture();
+//  char gesture = sensor.GetGesture();
 #pragma once
 
 #include <cstdint>
+#include <cstdlib>
+#include "L1_Drivers/i2c.hpp"
+#include "L2_Utilities/debug_print.hpp"
+
+I2c i2c(I2c::Port::kI2c2);
 
 class Apds9960Interface
 {
  public:
-        virtual void Initialize() = 0;
-        virtual void EnableGesture() = 0;
-        virtual void DisableGesture() = 0;
-        virtual void DecodeGestureData() = 0;
-        virtual uint8_t GetGesture() = 0;
-        virtual uint8_t ReadMode() = 0;
-        virtual uint8_t GetGestureEnterThreshhold() = 0;
-        virtual uint8_t GetGestureExitThreshhold() = 0;
-        virtual bool SetGestureEnterThresh(uint8_t threshhold_value) = 0;
-        virtual bool SetGestureExitThresh(uint8_t threshhold_value) = 0;
-        virtual bool CheckIfGestureOccured() = 0;
-        virtual void ReadGestureFIFO() = 0;
+  virtual void WriteReg(uint8_t reg, uint8_t data) = 0;
+  virtual uint8_t ReadReg(uint8_t reg) = 0;
+  virtual void Initialize() = 0;
+  virtual bool SetMode(uint8_t mode, bool enable) = 0;
+  virtual void EnableGesture() = 0;
+  virtual bool DisableGesture() = 0;
+  virtual uint8_t ReadGestureMode() = 0;
+  virtual bool CheckIfGestureOccured() = 0;
+  virtual uint8_t GetGestureFIFOLvl() = 0;
+  virtual void ReadGestureFIFO(uint8_t * data, uint8_t fifolvl) = 0;
+  virtual void ProcessGestureData(uint8_t level) = 0;
+  virtual char DecodeGestureData() = 0;
+  virtual char GetGesture() = 0;
+};
+
+class Apds9960 : public Apds9960Interface
+{
+ public:
+  // Apds-9960 Registers
+  static constexpr uint8_t
```

Comment:
I think should be within an `enum` but not an `enum class`. This wouldn't need to be strongly typed.